From d1aa02aa4f64a21e02d5aa18b16661226b7e421c Mon Sep 17 00:00:00 2001
From: Yanan Yang <yanan.yang@nxp.com>
Date: Thu, 26 Jun 2025 19:22:58 +0900
Subject: [PATCH 35/56] LF-15670: arm64: dts: freescale: add
 imx91-11x11-frdm-imx91s-ld.dts

Add low power control support for FRDM-IMX91S platform

Signed-off-by: Yanan Yang <yanan.yang@nxp.com>
Reviewed-by: Jacky Bai <ping.bai@nxp.com>
---
 arch/arm64/boot/dts/freescale/Makefile        |  3 +-
 .../freescale/imx91-11x11-frdm-imx91s-ld.dts  | 31 +++++++++++++++++++
 2 files changed, 33 insertions(+), 1 deletion(-)
 create mode 100644 arch/arm64/boot/dts/freescale/imx91-11x11-frdm-imx91s-ld.dts

diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index 630ce7a32c6d..59d2f5e48af7 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -521,7 +521,8 @@ dtb-$(CONFIG_ARCH_MXC) += imx91-11x11-evk.dtb \
 
 dtb-$(CONFIG_ARCH_MXC) += imx91-11x11-frdm-imx91s.dtb \
 			  imx91-11x11-frdm-imx91s-aud-hat.dtb \
-			  imx91-11x11-frdm-imx91s-8mic-revE.dtb
+			  imx91-11x11-frdm-imx91s-8mic-revE.dtb \
+			  imx91-11x11-frdm-imx91s-ld.dtb
 
 imx93-11x11-evk-pmic-pf0900-dtbs := imx93-11x11-evk.dtb imx93-11x11-evk-pmic-pf0900.dtbo
 imx93-11x11-evk-pmic-pf0900-root-dtbs := imx93-11x11-evk-root.dtb imx93-11x11-evk-pmic-pf0900.dtbo
diff --git a/arch/arm64/boot/dts/freescale/imx91-11x11-frdm-imx91s-ld.dts b/arch/arm64/boot/dts/freescale/imx91-11x11-frdm-imx91s-ld.dts
new file mode 100644
index 000000000000..15b685ae2b7f
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/imx91-11x11-frdm-imx91s-ld.dts
@@ -0,0 +1,31 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright 2025 NXP
+ */
+
+#include "imx91-11x11-frdm-imx91s.dts"
+
+&flexspi1 {
+	assigned-clocks = <&clk IMX91_CLK_FLEXSPI1>;
+	assigned-clock-parents = <&clk IMX91_CLK_SYS_PLL_PFD1>;
+	assigned-clock-rates = <266000000>;
+};
+
+&lpm {
+	no-od-mode;
+	ld-mode-enabled;
+	soc-supply = <&buck2>;
+	status = "okay";
+};
+
+&usdhc1 {
+	assigned-clocks = <&clk IMX91_CLK_USDHC1>;
+	assigned-clock-parents = <&clk IMX91_CLK_SYS_PLL_PFD1>;
+	assigned-clock-rates = <266000000>;
+};
+
+&usdhc2 {
+	assigned-clocks = <&clk IMX91_CLK_USDHC2>;
+	assigned-clock-parents = <&clk IMX91_CLK_SYS_PLL_PFD1>;
+	assigned-clock-rates = <266000000>;
+};
-- 
2.34.1

